src/Microsoft.NetCore.Analyzers/Core/Tasks/UseValueTasksCorrectly.cs

```diff
@@ -0,0 +1,547 @@
+            DiagnosticCategory.Reliability,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            s_localizableDescription,
+            helpLinkUri: null,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        internal static readonly DiagnosticDescriptor AccessingIncompleteResultDescriptor = new DiagnosticDescriptor(
+            RuleId,
+            s_localizableTitle,
+            new LocalizableResourceString(nameof(MicrosoftNetCoreAnalyzersResources.UseValueTasksCorrectlyMessage_AccessingIncompleteResult), MicrosoftNetCoreAnalyzersResources.ResourceManager, typeof(MicrosoftNetCoreAnalyzersResources)),
+            DiagnosticCategory.Reliability,
+            DiagnosticHelpers.DefaultDiagnosticSeverity,
+            isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
+            s_localizableDescription,
+            helpLinkUri: null,
+            customTags: WellKnownDiagnosticTags.Telemetry);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(GeneralDescriptor, UnconsumedDescriptor, DoubleConsumptionDescriptor, AccessingIncompleteResultDescriptor);
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                // Get the target ValueTask / ValueTask<T> types.  If they don't exist, nothing more to do.
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksValueTask, out var valueTaskType) ||
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemThreadingTasksGenericValueTask, out var valueTaskOfTType))
+                {
+                    return;
+                }
+
+                // Get the type for System.Diagnostics.Debug.  If we can't find it, that's ok, we just won't use it.
+                compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemDiagnosticsDebug, out var debugType);
+
+                // Process all invocations.  This analyzer works by finding all method invocations that return ValueTasks,
+                // and then analyzing those invocations and their surroundings.
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    // Is this a method invocation?  If not, we're done.
+                    var operation = operationContext.Operation;
+                    var invocation = (IInvocationOperation)operation;
+                    if (invocation.TargetMethod is null ||
```

Comment:
This if statement can be removed. `TargetMethod` is guaranteed to be non-null and not a constructor. Latter generates an `IObjectCreationOperation`.

Reply:
Thanks, @mavasani.  I thought that was probably the case, but I saw other analyzers doing this check, e.g.
https://github.com/dotnet/roslyn-analyzers/blob/2c46f6fe2e53b13b21066e1753cba95a8d6de3bb/src/Microsoft.NetCore.Analyzers/Core/Tasks/DoNotCreateTasksWithoutPassingATaskScheduler.cs#L57
and so wasn't sure.